modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/internal/data/handler/test/BatchEnginePortletDataHandlerTest.java

```diff
@@ -474,6 +485,270 @@ public void testExportImportPrivateLayoutsToOtherSite() throws Exception {
+			Collections.singletonMap("name", RandomTestUtil.randomString()));
+
+		objectEntryDtoAAAAA.setProperties(
+			Collections.singletonMap("name", RandomTestUtil.randomString()));
+
+		com.liferay.object.rest.dto.v1_0.ObjectEntry objectEntryAA =
+			defaultObjectEntryManager.addRelatedObjectEntry(
+				dtoConverterContext, objectEntryA.getExternalReferenceCode(),
+				objectEntryDtoAA, objectRelationshipA_AA, null);
+
+		com.liferay.object.rest.dto.v1_0.ObjectEntry objectEntryAAA =
+			defaultObjectEntryManager.addRelatedObjectEntry(
+				dtoConverterContext, objectEntryAA.getExternalReferenceCode(),
+				objectEntryDtoAAA, objectRelationshipAA_AAA, null);
+
+		com.liferay.object.rest.dto.v1_0.ObjectEntry objectEntryAAAA =
+			defaultObjectEntryManager.addRelatedObjectEntry(
+				dtoConverterContext, objectEntryAAA.getExternalReferenceCode(),
+				objectEntryDtoAAAA, objectRelationshipAAA_AAAA, null);
+
+		com.liferay.object.rest.dto.v1_0.ObjectEntry objectEntryAAAAA =
+			defaultObjectEntryManager.addRelatedObjectEntry(
+				dtoConverterContext, objectEntryAAAA.getExternalReferenceCode(),
+				objectEntryDtoAAAAA, objectRelationshipAAAA_AAAAA, null);
+
+		try {
+			File larFile = _exportLayouts(
+				false, group.getGroupId(), false, new long[0],
+				objectDefinitionA);
+
+			JSONArray exportedObjectEntriesJSONArray =
+				_getExportedObjectEntriesJSONArray(
+					objectDefinitionA.getName(), larFile, group.getGroupId());
+
+			JSONObject objectDefinitionJSONObject =
+				exportedObjectEntriesJSONArray.getJSONObject(0);
+
+			Assert.assertTrue(
+				objectDefinitionJSONObject.has(
+					objectRelationshipA_B.getName()));
+
+			objectDefinitionJSONObject = _assertAndGetJSONObject(
+				objectDefinitionJSONObject, objectRelationshipA_AA,
+				(String)objectEntryDtoAA.getPropertyValue("name"));
+
+			objectDefinitionJSONObject = _assertAndGetJSONObject(
+				objectDefinitionJSONObject, objectRelationshipAA_AAA,
+				(String)objectEntryDtoAAA.getPropertyValue("name"));
+
+			objectDefinitionJSONObject = _assertAndGetJSONObject(
+				objectDefinitionJSONObject, objectRelationshipAAA_AAAA,
+				(String)objectEntryDtoAAAA.getPropertyValue("name"));
+
+			_assertAndGetJSONObject(
+				objectDefinitionJSONObject, objectRelationshipAAAA_AAAAA,
+				(String)objectEntryDtoAAAAA.getPropertyValue("name"));
+
+			_deleteObjectEntries(objectEntryA);
+			_deleteObjectEntries(objectEntryB);
+
+			_importLayouts(
+				false, false, larFile, group.getGroupId(), objectDefinitionA);
+
+			com.liferay.object.rest.dto.v1_0.ObjectEntry objectEntryImported =
+				defaultObjectEntryManager.getObjectEntry(
+					objectDefinitionA.getCompanyId(), dtoConverterContext,
+					objectEntryA.getExternalReferenceCode(), objectDefinitionA,
+					objectDefinitionA.getScope());
+
+			Assert.assertNotNull(objectEntryImported);
+			Assert.assertEquals(
+				objectEntryA.getValues(
+				).get(
+					"name"
+				),
+				objectEntryImported.getPropertyValue("name"));
```

Comment:
Did you check whether `_assertObjectEntries()` could be reused here?
For example, each time you fetch the next `ObjectEntry`, you could call this method.

Also, did you consider adapting the existing assertion method to include the extra validations you need?
I noticed some repeated code, so I’m wondering if we can reuse the existing method and simply extend its assertions.